decision-record/DR007-form-is-live-first-then-add-draft.MD

```diff
@@ -0,0 +1,45 @@
+# DR007: We'll get ready to support a live form first, then build draft when we have time
+
+Date: 2022-08-01
+
+## Status
+
+Proposed
+
+## Context
+
+We need to know what is needed for the first form, so we know what work we need to prioritise. At the moment, in the development version of GOV.UK Forms, there is only one concept of a live form. There is no separate "draft" or preview version. There is no versioning. The way we would build the platform to support the first form would likely differ, depending on how we want to support the form creators at the Insolvency Service to make changes to their first live form. What happens to an end user and their data if they're in the middle of completing the form when the form creator makes a change?
```

Comment:
We need to agree on what option we will provide to our form creators to make their first form go live, so that we can prioritise the work for the upcoming sprints.

In the current development version of the GOV.UK Forms form-runner, there is no concept of a draft. So forms that are created and edited in the form-admin are displayed in preview on form-runner as 'live'.

This is not the intended technical implementation, as we'd like to be able to support 'daft' and 'live' state of a form, with potential for versioning. It is however a first pass option that will let us make the first form live quicker.

We are aware that the first (current) option will have an impact on the end user if a form creator makes direct edits to a live form and can potentially disrupt a form filling journey mid way. For example: What happens to an end user and their data if they're in the middle of completing the form when the form creator makes a change?

We are thinking that this is unlikely to happen for the first form, but we're will be looking into making that eventuality less disruptive.

We will be striving to design, test and build GOV.UK Forms to be more closely aligned with the option 4, where "A single “draft” version of a form can be worked on while a separate “live” version exists".

Reply:
@james-sheppard-gov-uk  is mentioning form-runner and admin too much?

Reply:
@deberny I suggest we keep the technical details out, just focus on the user experience.

Reply:
We've had a chat and agreed that technical bits are not needed. So I'll take my comment back.